core/gdb_console.c

```diff
@@ -21,23 +21,22 @@
 #include <core/gdb_console.h>
 
 
-static void writeStringToExclusiveGdbCommChannel(const char* pString);
 void WriteStringToGdbConsole(const char* pString)
 {
-    writeStringToExclusiveGdbCommChannel(pString);
+    WriteSizeStringToGdbConsole(pString, mri_strlen(pString));
 }
 
 /* Send the 'O' command to gdb to output text to its console.
 
     Command Format: OXX...
     Where XX is the hexadecimal representation of each character in the string to be sent to the gdb console.
 */
-static void writeStringToExclusiveGdbCommChannel(const char* pString)
+void WriteSizeStringToGdbConsole(const char* pString, size_t length)
 {
     Buffer* pBuffer = GetInitializedBuffer();
 
     Buffer_WriteChar(pBuffer, 'O');
-    Buffer_WriteStringAsHex(pBuffer, pString);
+    Buffer_WriteSizeStringAsHex(pBuffer, pString, length);
```

Comment:
Fails to compile this line as well. Should be Buffer_WriteSize**d**StringAsHex().